### PR TITLE
Directives to components

### PR DESCRIPTION
**Pull request by StevePole**

### CODE REVIEW

File: angular/form-components/tw-checkbox/tw-checkbox.component.js [hunk 1]
```diff
@@ -0,0 +1,130 @@
+
```
Comment (skywritergr) [inline angular/form-components/tw-checkbox/tw-checkbox.component.js]: space

File: angular/form-components/tw-checkbox/tw-checkbox.component.js [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+    bindings: {
+      name: "@",
+      ngModel: '=',
+      ngTrueValue: '<',
+      ngFalseValue: '<',
+      ngRequired: '<',
+      ngDisabled: '<'
+    },
+    template: " \
+      <input type='hidden' class='sr-only' \
```
Comment (skywritergr) [inline angular/form-components/tw-checkbox/tw-checkbox.component.js]: Let me say that this backslashes are annoying. With ES6 there are object literals that help you with this. Can you use them across your files?

https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Template_literals

File: angular/form-components/tw-checkbox/tw-checkbox.component.js [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+        aria-pressed='{{$ctrl.checked}}'> \
+        <span class='tw-checkbox-check glyphicon glyphicon-ok'></span> \
+      </button>"
+  };
+
+  angular
+    .module('tw.form-components')
+    .component('twCheckbox', TwCheckbox);
+
+  function TwCheckboxController($scope, $element) {
```
Comment (skywritergr) [inline angular/form-components/tw-checkbox/tw-checkbox.component.js]: This is an excellent client for starting using classes. We might want to start looking at setting up webpack

Comment (lapostoj) [inline angular/form-components/tw-checkbox/tw-checkbox.component.js]: Look at the name of the branch 😛

File: angular/form-components/tw-amount-currency-select/tw-amount-currency-select.directive.js [hunk 4]
```diff
@@ -1,91 +1,82 @@
...
-		.module('tw.form-components')
-		.directive('twAmountCurrencySelect', TwAmountCurrencySelectDirective);
+  const TwAmountCurrencySelect = {
+    require: 'ngModel',
+    controller: 'TwAmountCurrencySelectController',
+    transclude: {
+      'addon': '?addon'
+    },
+    bindings: {
+      ngModel: '=',
```
Comment (lapostoj) [inline angular/form-components/tw-amount-currency-select/tw-amount-currency-select.directive.js]: Just wondering, any plan to move completely to `<` later and notice of changes through callbacks or it feels a bit too much of an overhead for that kind of component?

**Comment by StevePole** [comment]
@oliverviljamaa Originally I changed the file names to .component.js but the PR was impossible to review.  I will make a separate PR with just renaming.

File: angular/form-components/tw-amount-currency-select/tw-amount-currency-select.directive.js (hunk 4 above)
Comment (StevePole) [inline angular/form-components/tw-amount-currency-select/tw-amount-currency-select.directive.js]: Would require massive changes across apps.  Would think we would need an extensive period of supporting both.

**Pull request merged**